generator/generator.md

```diff
@@ -0,0 +1,246 @@
+
+```typescript
+// home/widgets.tsx
+export const home_Q2_nbPersonnesResidenceTravailPreciser: inputTypes.InputString = {
+    ...defaultInputBase.inputNumberBase,
+    path: 'home.Q2_nbPersonnesResidenceTravailPreciser',
+    label: (t: TFunction) => t('home:home.Q2_nbPersonnesResidenceTravailPreciser'),
+    conditional: customConditionals.nbPersonnesSeptPlusConditional,
+    validations: validations.moreOrEqualTo7Validation
+};
+```
+
+### Widgets Example
+![Widgets example](./assets/images/widgetsExample.png)
+
+## Generate Conditionals
+
+In your survey logic, conditionals play a key role in determining if the widget will appear or not. The table below outlines the fields used to define conditionals in Conditionals tab, along with an example and the corresponding TypeScript code.
+
+### Conditionals Fields
+
+| Field                       | Description                            | Example                       | Type                    |
+|-----------------------------|----------------------------------------|-------------------------------|-------------------------|
+| conditionalName             | Name of the conditional                | siPermisConduireConditional   | string                  |
+| logicalOperator             | Logical operator (optional)            | &&                            | && or \|\|              |
+| path                        | Path to the data point for comparison  | [${relativePath}](#rel).age   | string                  |
+| [comparisonOperator](#comp) | Operator for comparison                | >=                            | ===, ==, >, <, >= or <= |
+| value                       | Value for the comparison               | 16                            | number or string        |
+| [parentheses](#par)         | Parentheses (optional)                 |                               | ( or )                  |
+
+> <span id="comp">**Note:**</span> The `comparisonOperator` field helps compare respondent responses with the specified value. It determines how the respondent's answer should be evaluated in the conditional logic. In this example, `>=` signifies that the condition is true when `person.age` is greater than or equal to 16.
+
+> <span id="par">**Note:**</span> The `parentheses` field is optional and allows you to add priority to the conditional logic by using opening and closing parentheses. This is useful for specifying the order in which conditions should be evaluated. For example, you can use parentheses to create complex conditions like `conditional1 || (conditional2 && conditional3)`, where `conditional2 && conditional3` is evaluated first due to the parentheses.
+
+> <span id="rel">**Note:**</span> `${relativePath}` in `path` is used to obtain the relative path within the same group, facilitating the reference to data points that share a common parent or group with the current data point being processed in the TypeScript code for generating conditionals.
+
+### Conditionals TypeScript
```

Comment:
Here you should put an example csv data with what is required to generate the code below, with exact fields. That will be more helpful for the documentation readers

Reply:
Great idea.